docs/usage.rst

```diff
@@ -0,0 +1,198 @@
+.. _usage:
+
+Usage
+=====
+Preparing your WDL files
+------------------------
+Before the documentation can be generated, each input in your WDL_ file must
+be given a description and category. This is done using WDL's parameter_meta_
+section:
+
+.. code-block:: javascript
+
+    parameter_meta {
+        name_of_input: {
+            description: "Some description of the input",
+            category: "some category"
+        }
+    }
+
+These fields (``description`` and ``category`` may also be called differently,
+but you will have to set some additional options when running WDL-AID, see
+`Custom description and category keys`_.
+
+WDL-AID will separate the inputs by category, so each category may be rendered
+in its own section. Required inputs are automatically detected and assigned
+the ``required`` category, overwriting the one noted in the parameter_meta
+section.
+
+The default template supports the following categories:
+
+- ``required``
+- ``common``
+- ``advanced``
+- ``other``
+
+Excluding inputs
+^^^^^^^^^^^^^^^^
+In some cases there may be inputs which should not be included in the
+documentation, eg. when using a sub-workflow which provides options which make
+no sense in the context of the overarching workflow. You can tell WDL_AID to
+omit certain inputs by adding the following to your workflow's meta_ section:
+
+.. code-block:: javascript
+
+    WDL_AID: {
+        exclude: ["input_name", "call.input_name"]
+    }
+
+The inputs added here may be of the workflow or task containing the meta section
+or from any call made inside of the workflow. Be sure to use the input_names
+qualified relative to this workflow, ie. ``input_name`` for inputs of the
+task/workflow itself, ``call_name.input_name`` for inputs of calls,
+``call_name.sub_call_name.input_name`` for calls inside of sub-workflows, etc.
+
+Metadata
+^^^^^^^^
+The meta_ section of your workflow may also be used to pass additional
+information along to WDL-AID. The entire meta section of the workflow
+that WDL-AID gets called on is passed to the template unaltered.
+WDL-AID will also retrieve all authors from the meta sections of every called
+task and workflow and pass these along.
+
+The default template supports the following meta section entries:
+
+- ``description`` (only for the root workflow)
+- ``authors`` (also for sub-workflows and tasks)
+
+    This is expected to be an object containing the following
+    fields for each author:
+
+    - ``name``
+    - ``email`` (optional)
+    - ``organization`` (optional)
+
+    eg.
+
+    .. code-block:: javascript
+
```

Comment:
WDL not yet supported by pygments? Oh well. Javascript is indeed a good alternative.